### PR TITLE
[release/6.0] Install native dependencies on all Apple OS in CI

### PR DESCRIPTION
**Pull request by am11**
Backport of https://github.com/dotnet/runtime/pull/82263.

### CODE REVIEW

File: eng/install-native-dependencies.sh
Comment (am11) [inline eng/install-native-dependencies.sh:15]: This line is different; lowercase changes in 8.0 didn't require it.

Comment (am11) [inline eng/install-native-dependencies.sh:15]: This line is new; lowercase changes in 8.0 didn't require this transformation.

**Review by steveisok — APPROVED** [review]

**Comment by akoeplinger** [comment]
https://github.com/actions/runner-images/pull/7125 was merged which should add pkg-config back, it still needs to roll out as builds still fail as of right now. I'd be in favor of merging this but I'll let @carlossanlop decide.

**Comment by am11** [comment]
We run `install-native-dependencies` is some OSX legs, this PR is basically aligning it to run on all OSX legs with native build. The installation is idempotent; it only installs the missing dependencies. This step is taking 15-20 seconds.

**Comment by steveisok** [comment]
@carlossanlop are you good with merging this?

**Comment by carlossanlop** [comment]
Waiting for Matt's answer here: https://github.com/dotnet/runtime/pull/82323#issuecomment-1442681623

**Pull request merged**
